samples/testapp/src/commonMain/kotlin/com/android/identity/testapp/ui/IsoMdocProximitySharingScreen.kt

```diff
@@ -167,339 +93,171 @@ fun IsoMdocProximitySharingScreen(
-        Logger.i(TAG, "Ending holderJob due to MdocTransportClosedException")
-    } catch (error: Throwable) {
-        Logger.e(TAG, "Caught exception", error)
-        error.printStackTrace()
-        showToast("Error: ${error.message}")
-    } finally {
-        transport.close()
-        holderTransport.value = null
-        encodedDeviceEngagement.value = null
     }
-}
-
-// Returns consent-fields if confirmed by the user, false otherwise.
-//
-// Throws if there is no mDL docrequest.
-//
-@OptIn(ExperimentalResourceApi::class)
-private suspend fun showConsentPrompt(
-    encodedDeviceRequest: ByteArray,
-    encodedSessionTranscript: ByteArray,
-    holderConsentSheetData: MutableState<ConsentSheetData?>,
-): List<MdocConsentField>? {
-    val deviceRequest = DeviceRequestParser(
-        encodedDeviceRequest,
-        encodedSessionTranscript,
-    ).parse()
-    for (docRequest in deviceRequest.docRequests) {
-        Logger.i(TAG, "docRequest.readerAuthenticated=${docRequest.readerAuthenticated}")
-        val trustPoint = if (docRequest.readerAuthenticated) {
-            val trustResult = TestAppUtils.readerTrustManager.verify(docRequest.readerCertificateChain!!.certificates)
-            Logger.i(TAG, "trustResult.isTrusted=${trustResult.isTrusted}")
-            Logger.i(TAG, "trustResult.error=${trustResult.error}")
-            if (trustResult.isTrusted) {
-                trustResult.trustPoints[0]
-            } else {
-                null
-            }
-        } else {
-            null
-        }
-        if (docRequest.docType == DrivingLicense.MDL_DOCTYPE) {
-            val cardArt = getDrawableResourceBytes(
-                getSystemResourceEnvironment(),
-                Res.drawable.driving_license_card_art
-            )
-            val consentFields = MdocConsentField.generateConsentFields(
-                docRequest,
-                TestAppUtils.documentTypeRepository,
-                TestAppUtils.mdocCredential
-            )
-            if (suspendCancellableCoroutine { continuation ->
-                holderConsentSheetData.value = ConsentSheetData(
-                    showConsentPrompt = true,
-                    continuation = continuation,
-                    document = ConsentDocument(
-                        name = "Erika's Driving License",
-                        cardArt = cardArt,
-                        description = "Driving License",
-                    ),
-                    consentFields = consentFields,
-                    relyingParty = ConsentRelyingParty(
-                        trustPoint = trustPoint,
-                        websiteOrigin = null,
-                    )
-                )
-            }) {
-                return consentFields
-            } else {
-                return null
-            }
-
-        }
+    // Wait until state changes to CONNECTED, CONNECTING, FAILED, or CLOSED
+    transport.state.first {
```

Comment:
Similar comment to the other state flow first() call: assign to a val and check that val? And rename state to stateFlow?

Reply:
Will address in next PR (this code will be moved so it's easier to edit it in the next PR)